### ISSUE TITLE
Get message ID after sending a message

### ISSUE THREAD

**Issue by SiliconDroid** [iotaledger/iota.c#135]
After using wallet_send, I want to store the message ID.

I have modified wallet_send to accept a byte buffer in which the sent message ID is copied:

wallet.h:

```c
/**
 * @brief Send message to the Tangle
 *
 * @param[in] w A wallet instance
 * @param[in] sender_index The address index of this wallet
 * @param[in] receiver The receiver address in ed25519 format
 * @param[in] balance The balance to send
 * @param[in] index An optional indexation
 * @param[in] data An optional indexation data, it's ignored if the index parameter is NULL
 * @param[in] data_len String length of the optional indexation data
 * @param[in] sent_message_id An optional byte buffer to receive the resulting message id string (must be API_MSG_ID_HEX_BYTES+1 long)
 * @return int 0 on success
 */
int wallet_send(iota_wallet_t* w, uint32_t sender_index, byte_t receiver[], uint64_t balance, char const index[],
                byte_t data[], size_t data_len, byte_t sent_message_id[]);
```

wallet.c:

```c
  // send message
  if (send_core_message(&w->endpoint, msg, &msg_res) == 0) {
    printf("[%s:%d] message ID: %s\n", __func__, __LINE__, msg_res.u.msg_id);

    // FKA_TAG 26. May 2021 12:34 ----------------------------------------
    if(sent_message_id)
    {
      memccpy(sent_message_id, msg_res.u.msg_id, 0, API_MSG_ID_HEX_BYTES);
    }
    // FKA_TAG 26. May 2021 12:34 ----------------------------------------

    core_message_free(msg);
```

I'm not sure if you can think of a better way of doing this? Perhaps add a function that returns some static byte array used to remember the last ID.

```c
byte_t[] wallet_get_last_sent_message_id()
```

This way it adds the fucntionality without being a breaking change?

**Comment by oopsmonk** [comment]
Wallet APIs are WIP, adding an extra parameter to store the message ID is a good idea.

**Comment by oopsmonk** [comment]
@SiliconDroid  fixed at 8461ca2e7a4f782e29ffcbe6da9674d38f17ebb9
